pkg/libovsdb/encoding_test.go

```diff
@@ -2,48 +2,214 @@ package libovsdb
+	{
+		objInput:           []string{},
+		jsonExpectedOutput: `["set",[]]`,
+	},
+	{
+		objInput:           `aa`,
+		jsonExpectedOutput: `"aa"`,
+	},
+	{
+		objInput:           false,
+		jsonExpectedOutput: `false`,
+	},
+	{
+		objInput:           10,
+		jsonExpectedOutput: `10`,
+	},
+	{
+		objInput:           10.2,
+		jsonExpectedOutput: `10.2`,
+	},
+	{
+		objInput:           []string{`aa`},
+		jsonExpectedOutput: `"aa"`,
+	},
+	{
+		objInput:           [1]string{`aa`},
+		jsonExpectedOutput: `"aa"`,
+	},
+	{
+		objInput:           []string{`aa`, `bb`},
+		jsonExpectedOutput: `["set",["aa","bb"]]`,
+	},
+	{
+		objInput:           [2]string{`aa`, `bb`},
+		jsonExpectedOutput: `["set",["aa","bb"]]`,
+	},
+	{
+		objInput:           []int{10, 15},
+		jsonExpectedOutput: `["set",[10,15]]`,
+	},
+	{
+		objInput:           []uint16{10, 15},
+		jsonExpectedOutput: `["set",[10,15]]`,
+	},
+	{
+		objInput:           []float32{10.2, 15.4},
+		jsonExpectedOutput: `["set",[10.2,15.4]]`,
+	},
+	{
+		objInput:           []float64{10.2, 15.4},
+		jsonExpectedOutput: `["set",[10.2,15.4]]`,
+	},
+	{
+		objInput:           []UUID{},
+		jsonExpectedOutput: `["set",[]]`,
+	},
+	{
+		objInput:           UUID{GoUUID: `aa`},
+		jsonExpectedOutput: `["named-uuid","aa"]`,
+	},
+	{
+		objInput:           []UUID{{GoUUID: `aa`}},
+		jsonExpectedOutput: `["named-uuid","aa"]`,
+	},
+	{
+		objInput:           []UUID{{GoUUID: `aa`}, {GoUUID: `bb`}},
+		jsonExpectedOutput: `["set",[["named-uuid","aa"],["named-uuid","bb"]]]`,
+	},
+	{
+		objInput:           validUUID0,
+		jsonExpectedOutput: fmt.Sprintf(`["uuid","%v"]`, validUUIDStr0),
+	},
+	{
+		objInput:           []UUID{validUUID0},
+		jsonExpectedOutput: fmt.Sprintf(`["uuid","%v"]`, validUUIDStr0),
+	},
+	{
+		objInput:           []UUID{validUUID0, validUUID1},
+		jsonExpectedOutput: fmt.Sprintf(`["set",[["uuid","%v"],["uuid","%v"]]]`, validUUIDStr0, validUUIDStr1),
+	},
+}
+
+var mapTestList = []marshalMapsTestTuple{
+	{
+		objInput:           map[string]string{},
+		jsonExpectedOutput: `["map",[]]`,
+	},
+
+	{
+		objInput:           map[string]string{`v0`: `k0`},
+		jsonExpectedOutput: `["map",[["v0","k0"]]]`,
+	},
+
+	{
+		objInput:           map[string]string{`v0`: `k0`, `v1`: `k1`},
+		jsonExpectedOutput: `["map",[["v0","k0"],["v1","k1"]]]`,
+	},
+}
+
+// Json array is not order sensitive, but Golang set is, so we have to compare teh sets independently to the order of
+// its elements
+func setsAreEqual(t *testing.T, set1 *OvsSet, set2 *OvsSet) {
+	res1 := map[interface{}]bool{}
+	for _, elem := range set1.GoSet {
+		switch elem.(type) {
+		case UUID:
+			uuid := elem.(UUID)
+			res1[uuid.GoUUID] = true
+		default:
+			s := fmt.Sprintf("%v", elem)
+			res1[s] = true
+		}
+	}
+
+	res2 := map[interface{}]bool{}
+	for _, elem := range set2.GoSet {
+		switch elem.(type) {
+		case UUID:
+			uuid := elem.(UUID)
+			res2[uuid.GoUUID] = true
+		default:
+			s := fmt.Sprintf("%v", elem)
+			res2[s] = true
+		}
+	}
```

Comment:
this is a total code duplication of the block above, please use functions ...

Reply:
the function was removed